samcli/cli/context.py

```diff
@@ -32,7 +32,7 @@ def __init__(self):
         Initialize the context with default values
         """
         self._debug = False
-        self._aws_region = None
+        self._aws_region = boto3.session.Session().region_name
```

Comment:
botocore.session.get_session() might be better, this might create a brand new session.